### DISCUSSION TITLE
Make the regex module simpler

### DISCUSSION THREAD

**Discussion by Olyno** [vlang/v#12693] [Ideas]
Hi :wave:,

I've needed the regex module several times during my tests with V, and to be quite honest, I find it too complicated to use and not very intuitive. 

One of the advantages of the V language is its simplicity, and I never had this problem with the other modules.

So I suggest a small redesign of the api. The goal would be to make the regex module as simple as possible, as simple as the other modules. Based on what already exists, here is an suggestion I can give:

**Before:**

```v
text := 'cpaz cpapaz cpapapaz'
query := r'(c(pa)+z ?)+'
mut re := regex.regex_opt(query) or { panic(err) }
println(re.get_query())
// #0(c#1(pa)+z ?)+
// #0 and #1 are the ids of the groups, are shown if re.debug is 1 or 2
start, end := re.match_string(text)
// [start=0, end=20]  match => [cpaz cpapaz cpapapaz]
mut gi := 0
for gi < re.groups.len {
    if re.groups[gi] >= 0 {
        println('${gi / 2} :[${text[re.groups[gi]..re.groups[gi + 1]]}]')
    }
    gi += 2
}
// groups captured
// 0 :[cpapapaz]
// 1 :[pa]
```

**After:**

```v
text := 'cpaz cpapaz cpapapaz'
query := r'(c(pa)+z ?)+'
groups := re.match(text)
// groups : [ RegMatch{ start: 0, end: 20, text: "cpapapaz" } ]
```

I think it can be good to have simpler methods like in javascript for example (the list of existing methods available [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#using_regular_expressions_in_javascript))

Any feedback about it is welcome 😄

**Comment by penguindark** [comment]
Interesting idea, but this simple return is not possible if you do not return a `struct`.
Having a wrapper is always possible.
